Lib/test/test__xxsubinterpreters.py

```diff
@@ -1337,6 +1337,97 @@ def test_run_string_arg_resolved(self):
         self.assertEqual(obj, b'spam')
         self.assertEqual(out.strip(), 'send')
 
+    def test_channel_send_wait_same_interpreter(self):
+        cid = interpreters.channel_create()
+
+        received = interpreters.channel_send_wait(cid, b"send", timeout=0)
+        self.assertFalse(received)
+
+        obj = interpreters.channel_recv(cid)
+        self.assertEqual(obj, b"send")
+
+    def test_channel_send_wait_different_interpreters(self):
+        cid = interpreters.channel_create()
+        interp = interpreters.create()
+        _run_output(interp, dedent(f"""
+            import _xxsubinterpreters as _interpreters
+            import time
+            import math
+
+            start = time.time()
+            rc = _interpreters.channel_send_wait({cid}, b"send", timeout=1)
```

Comment:
It would be worth making it clear (e.g. in a comment) that we expect this to time out (since it is happening in the same thread where we expect `recv()` to be called later).